src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1192,29 +1199,81 @@ protected override void ApplyAnalyzerConfigDocumentTextChanged(DocumentId docume
-            TextEditApplication.UpdateText(newText, invisibleEditor.TextBuffer, EditOptions.None);
+                // The document wasn't open in a normal way, so invisible editor time
+                using var invisibleEditor = OpenInvisibleEditor(documentId);
+                TextEditApplication.UpdateText(newText, invisibleEditor.TextBuffer, EditOptions.None);
+            }
         }
+        catch (Exception ex) when (FatalError.ReportAndCatch(ex))
+        {
+            ReportErrorChangingDocumentText(ex);
+            return;
+        }
+
+        void ReportErrorChangingDocumentText(Exception exception)
+        {
+            // Don't spam the info bar.  If the user already has a message up, leave it at that.  Also,
+            // if they've asked to not be notified about future doc change issue, respect that flag.
+            if (_ignoreDocumentTextChangeErrors || _isShowingDocumentChangeErrorInfoBar)
+                return;
+
+            var documentName = document?.Name ?? documentId.ToString();
+
+            var errorReportingService = this.Services.GetRequiredService<IErrorReportingService>();
+            errorReportingService.ShowGlobalErrorInfo(
+                message: string.Format(ServicesVSResources.Error_encountered_updating_0, documentName),
+                TelemetryFeatureName.Workspace,
+                exception,
+                // 'Show stack trace' will not dismiss the info bar.
+                new(WorkspacesResources.Show_Stack_Trace, InfoBarUI.UIKind.HyperLink,
+                    () => errorReportingService.ShowDetailedErrorInfo(exception), closeAfterAction: false),
+                // 'Ignore' just closes the info bar, but allows future errors to show up.
+                new(ServicesVSResources.Ignore, InfoBarUI.UIKind.Button, GetDefaultDismissAction()),
+                // 'Ignore (including future errors) closes the info bar, but also sets the flag so the user gets no more messages
+                // in the current session.
+                new(ServicesVSResources.Ignore_including_future_errors, InfoBarUI.UIKind.Button, GetDefaultDismissAction(
+                    () => _ignoreDocumentTextChangeErrors = true)),
+                // Close button is the same as 'ignore'.  It closes the info bar, but allows future errors to show up.
+                new(string.Empty, InfoBarUI.UIKind.Close, GetDefaultDismissAction()));
+
+            // Mark that we're showing the info bar at this point.
+            _isShowingDocumentChangeErrorInfoBar = true;
+        }
+
+        Action GetDefaultDismissAction(Action? additionalAction = null)
+            => () =>
+            {
+                additionalAction?.Invoke();
+
+                // All info bar actions (except for 'show stack trace') dismiss the info bar, putting us back in the
+                // "we're not showing the user anything" state.
+                _isShowingDocumentChangeErrorInfoBar = false;
```

Comment:
Too bad there's not a generic way for us just to pass a handler for "any way this goes away".... :-/